src/rebar_prv_local_upgrade.erl

```diff
@@ -52,7 +52,24 @@ do(State) ->
 
             case maybe_fetch_rebar3(Md5) of
                 {saved, TmpRebar3} ->
-                    rebar_prv_local_install:extract_escript(State, TmpRebar3);
+                    {Vsn, Archive} =
+                        try
+                            {ok, Escript} = escript:extract(TmpRebar3, []),
+                            {archive, FullArchive} = lists:keyfind(archive, 1, Escript),
+                            {ok, [{_, AppFile}]} = zip:extract(FullArchive,
+                                                               [memory,
+                                                                {file_list, ["rebar/ebin/rebar.app"]}]),
+
+                            {ok, {application, _Name, AppDetails}} = rebar_string:consult(AppFile),
```

Comment:
`rebar_string:consult/1` just returns a list of terms or the error tuple, it appears we messed up the types and didn't wrap it in `{ok, [T]}`. 

There's also the option I mentioned in the issue of just looking for the comment on the second line of the archive, which should contain a comment, not sure if that's somewhat safer?

Reply:
Oh right, the version in the comment at the top of the escript... I misunderstood what you had suggested earlier since I forgot we put that there. That may be better. I don't really like string parsing/regex from a file vs reading a term, but will try it.

Reply:
Nevermind, this is definitely better. I didn't realize there was the `comment` field as part of the extraction.